### PR TITLE
Added ML clustering tutorial

### LINKED ISSUE #5936: Clustering Iris tutorial is missing
Where is the tutorial about the Clustering Iris , i found the tutorial in dot.net/ml but there wasn't any code explanation like the Sentiment Analysis and Taxi Fare Predictor tutorials

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e855e630-1a30-d138-4ac6-bfeefebf9aae
* Version Independent ID: afc0a7d1-d3c2-acf2-d4ea-d2565e22b237
* Content: [ML.NET tutorials](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/index#feedback)
* Content Source: [docs/machine-learning/tutorials/index.md](https://github.com/dotnet/docs/blob/master/docs/machine-learning/tutorials/index.md)
* Product: **dotnet-ml**
* GitHub Login: @JRAlexander
* Microsoft Alias: **johalex**

### PR DESCRIPTION
**Pull request by pkulikov**
This PR adds iris clustering tutorial to illustrate usage of the clustering ML task.

The structure is similar to the taxi fare tutorial.
For the code, kudos to @OliaG (I've used [the code from the machine learning samples repo](https://github.com/dotnet/machinelearning-samples/tree/master/samples/getting-started/Clustering_Iris) as base)

Supported by the source code in dotnet/samples#141

Fixes #5936 

@JRAlexander @OliaG @GalOshri please review

/cc: @Amine-Smahi

### CODE REVIEW

**Review by OliaG — APPROVED** [review]
This is amazing! Thank you very much @pkulikov. I'd ask @JRAlexander or @mairaw to review the language, ML-wise everything is great!

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+1. Install the **Microsoft.ML** NuGet package:
+
+    In **Solution Explorer**, right-click the project and select **Manage NuGet Packages**. Choose "nuget.org" as the Package source, select the **Browse** tab, search for **Microsoft.ML**, select that package in the list, and select the **Install** button. Select the **OK** button on the **Preview Changes** dialog and then select the **I Accept** button on the **License Acceptance** dialog if you agree with the license terms for the packages listed.
+
+## Prepare the data
+
+1. Download the [iris.data](https://github.com/dotnet/machinelearning/blob/master/test/data/iris.data) data set and save it to the *Data* folder you've created at the previous step. For more information about the iris data set, see the [Iris flower data set](https://en.wikipedia.org/wiki/Iris_flower_data_set) Wikipedia page and the [Iris Data Set](http://archive.ics.uci.edu/ml/datasets/Iris) page, which is the source of the data set.
+
+1. In **Solution Explorer**, right-click the *iris.data* file and select **Properties**. Under **Advanced**, change the value of **Copy to Output Directory** to **Copy always**.
```
[the remark this answers is not in the document]
Comment (pkulikov) [reply docs/machine-learning/tutorials/iris-clustering.md]: @OliaG that's true. I've also applied it to two other tutorials, but in different PR. Thank you.

**Comment by JRAlexander** [comment]
Reviewing. Just FYI.

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
+---
+title: Use ML.NET to cluster iris flowers (clustering)
+description: Learn how to use ML.NET in a clustering scenario
+author: pkulikov
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: Please add ms.author: johalex. That is for support routing.

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+}
+```
+
+The learning pipeline loads all of the data and algorithms necessary to train the model. Add the following code into the `Train` method:
+
+[!code-csharp[Initialize pipeline](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#5)]
+
+## Load and transform data
+
+The first step to perform is to load the training data set. In our case, training data set is stored in the text file with a path defined by the `_dataPath` field. Columns in the file are separated by the comma (","). Add the following code into the `Train` method:
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: "In our case, training data set" -> missing "the"

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+[!code-csharp[Add step to load data](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#6)]
+
+The next step is to combine all of the feature columns into the **Features** column using the <xref:Microsoft.ML.Transforms.ColumnConcatenator> transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
+
+[!code-csharp[Add step to concatenate columns](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#7)]
+
+## Choose a learning algorithm
+
+After adding the data to the pipeline and transforming it into the correct input format, you select a learning algorithm (**learner**). The learner trains the model. ML.NET provides a <xref:Microsoft.ML.Trainers.KMeansPlusPlusClusterer> learner that implements [k-means algorithm](https://en.wikipedia.org/wiki/K-means_clustering) with improved method for choosing the initial cluster centroids.
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: "with **an** improved method"

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+Add the following code into the `Train` method following the data processing code added in the preceding section:
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: "previous step" is more consistent than "preceding section".

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+[!code-csharp[Add a learner step](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#8)]
+
+Note that we specify that the data set should be split in three clusters.
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: Please state where is this specified.

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+
+## Train the model
+
+The steps added in the preceding sections prepared the pipeline for training. None of the steps has been executed. The `pipeline.Train<TInput, TOutput>` method produces the model that takes in an instance of the `TInput` type and outputs an instance of the `TOutput` type. Add the following code into the `Train` method:
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: The steps added in the preceding sections prepared the pipeline for training, however, none have been executed.

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+
+[!code-csharp[Train the model and return](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#9)]
+
+### Save the model
+
+Before you go onto the next step, save the model to a .zip file. To do that add the following code to the `Main` method below the call to the `Train` method:
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: Explain why we are saving the model.  "At this point, you have a model that can be integrated into any of your existing or new .NET applications. To save your model to a .zip file, add the following code to the `Main` method below the call to the `Train` method:"

File: docs/machine-learning/tutorials/iris-clustering.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+
+This tutorial introduces one iris data instance within this class. You can add other scenarios to experiment with the model. Add the following code into the `TestIrisData` class:
+
+[!code-csharp[Test data](../../../samples/machine-learning/tutorials/IrisClustering/TestIrisData.cs#2)]
+
+To find out the cluster to which the specified item belongs to, go back to the *Program.cs* file and add the following code into the `Main` method:
+
+[!code-csharp[Predict and output results](../../../samples/machine-learning/tutorials/IrisClustering/Program.cs#13)]
+
+Run the program to see which cluster contains the specified data instance and squared distances from that instance to the cluster centroids.
```
Comment (JRAlexander) [inline docs/machine-learning/tutorials/iris-clustering.md]: Please show the output like this [example in the Sentiment Analysis tutorial](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/sentiment-analysis#results).

**Review by JRAlexander — CHANGES_REQUESTED** [review]
Thanks, @pkulikov, for this great tutorial. There are just a few items to change and then we should be ready.

File: docs/machine-learning/tutorials/iris-clustering.md (hunk 8 above)
Comment (pkulikov) [reply docs/machine-learning/tutorials/iris-clustering.md]: I've also updated the taxi fare tutorial with this explanation. Thanks!

**Comment by pkulikov** [comment]
@JRAlexander thank you for the review! I've addressed your feedback.

**Review by JRAlexander — APPROVED** [review]

**Comment by JRAlexander** [comment]
Closing and reopening to restart build

**Pull request merged**